src/frontend/src/pages/TeamsPage/TeamMembersPageBlock.tsx

```diff
@@ -30,17 +30,21 @@ const TeamMembersPageBlock: React.FC<TeamMembersPageBlockProps> = ({ team }) =>
   const auth = useAuth();
   const [isEditingMembers, setIsEditingMembers] = useState(false);
   const [isEditingHead, setIsEditingHead] = useState(false);
+  const [isEditingLeads, setIsEditingLeads] = useState(false);
   const [members, setMembers] = useState(team.members.map(userToAutocompleteOption));
   const [head, setHead] = useState(userToAutocompleteOption(team.head));
+  const [leads, setLeads] = useState(team.leads.map(userToAutocompleteOption));
 
   const { isLoading: allUsersIsLoading, isError: allUsersIsError, error: allUsersError, data: users } = useAllUsers();
   const { isLoading: setTeamMembersIsLoading, mutateAsync: setTeamMembersMutateAsync } = useSetTeamMembers(team.teamId);
   const { isLoading: setTeamHeadIsLoading, mutateAsync: setTeamHeadMutateAsync } = useSetTeamHead(team.teamId);
+  const { isLoading: setTeamLeadsIsLoading, mutateAsync: setTeamLeadsMutateAsync } = useSetTeamLeads(team.teamId);
 
   const toast = useToast();
 
   if (allUsersIsError) return <ErrorPage message={allUsersError?.message} />;
-  if (allUsersIsLoading || setTeamMembersIsLoading || setTeamHeadIsLoading || !users) return <LoadingIndicator />;
+  if (allUsersIsLoading || setTeamMembersIsLoading || setTeamHeadIsLoading || setTeamLeadsIsLoading || !users)
+    return <LoadingIndicator />;
 
   const hasPerms = auth.user && (isAdmin(auth.user.role) || auth.user.userId === team.head.userId);
```

Comment:
not on you, but can you update this file to use useCurrentUser rather than useAuth (might have to update some tests as well)